prow/tide/history/history.go

```diff
@@ -105,9 +225,9 @@ func (h *History) AllRecords() map[string][]*Record {
 
 // recordLog is a space efficient, limited size, append only list.
 type recordLog struct {
-	buff  []*Record
-	head  int
-	limit int
+	Buff  []*Record
```

Comment:
I used pointers to avoid copying the entire Record structs around. e.g. the cachedSlice contains the same Records, but in order and doesn't have to double the memory footprint to do so.